docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +279,203 @@ Note that:
+       `tedge/$CHILD_DEVICE_ID/commands/res/config_snapshot`
+    1. There is three different payloads to notify the configuration operation started,
+       succeed or failed for some reason.
+    1. All these payloads are JSON records with 3 required fields:
+        * `"status": "$STATUS"` where the status is either "executing", "successful" or "failed"
+        * `"path": "$PATH"`
+        * `"type": "$TYPE"`
+        * `"reason": "$ERROR_MSG"` telling the cause of the error if any.
+    1. The child-device agent must send at least a success or an error message,
+       depending on the success of the `PUT` operation.
+       It should also send an executing message before starting to process the request.
+1. On reception of an operation status message,
+   the `c8y_configuration_plugin` notifies the cloud accordingly.
+    1. When a success message is received,
+       then the configuration plugin transfers to the cloud the content `PUT` by the child-device
+       under `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshop/$TYPE` and
+       finally removes this file when acknowledged by the cloud.
+    1. If no responses are received from the child devices after 60s,
+       then a timeout error is sent to the cloud,
+       and the `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshop/$TYPE` file removed
+       (if any has been uploaded but in an unknown state).
+    1. If a notification message is received while none is expected for this type of configuration
+       (this might notably arrive after a timeout), then this notification message is ignored.
+       
+### The child device uploads its configuration file list on start and on change
+
+The configuration files for a child device `$CHILD_DEVICE_ID` are defined in
+the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` on the thin-edge device.
+But, the content of this file must be provided by the child-device agent
+by uploading a `c8y-configuration-plugin.toml` with `"type": "c8y-configuration-plugin"` in its prescribed format.
+
+The child-device must upload this file on its startup as well as when updated,
+as if it received a config snapshot request for `c8y-configuration-plugin` type as follows:
+
+1. On start-up, the child-device uploads this file (as well as on update of this file list),
+   as if it received a config snapshot request for `c8y-configuration-plugin` type:
+      1. Generate a `c8y-configuration-plugin.toml` with the supported config list in the presribed format.
+      2. Uploads this file to `http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_snapshot/c8y-configuration-plugin` with a `PUT` call.
+      3. On success of the upload, the child-device agent
+         notifies the `c8y_configuration_plugin` with an MQTT message published on the topic
+         `tedge/$CHILD_DEVICE_ID/commands/res/config_snapshot`
+         with the payload containing a JSON record with 2 required fields:
+        * `"status": "successful"`
```

Comment:
Here is an idea that might simplify a lot the code of the `c8y_configuration_plugin`.

```suggestion
        * `"status": "updated"`
```

By having the child-device sending a specific status for spontaneous snapshot, this helps the main device to figure out that nothing is expected from the cloud.

Reply:
Instead of a new `status` value, we've decided to not have the `status` field for such spontaneous responses.